test-data/pybind11_mypy_demo/src/main.cpp

```diff
@@ -134,14 +140,18 @@ void bind_basics(py::module& basics) {
     .def(py::init<double, double>(), py::arg("x"), py::arg("y"))
     .def("distance_to", py::overload_cast<double, double>(&Point::distance_to, py::const_), py::arg("x"), py::arg("y"))
     .def("distance_to", py::overload_cast<const Point&>(&Point::distance_to, py::const_), py::arg("other"))
-    .def_readwrite("x", &Point::x)
+    .def("as_vector", &Point::as_vector)
+    // Note that the trailing newline is required because the generated docstring
+    // is concatenated to the signature, e.g.:
+    // 'some docstring\n(self: pybind11_mypy_demo.basics.Point) -> float\n'
```

Comment:
Can this not be solved internally? Or why did mypy <1.7 not have this quirk?

This is worrying, because users cannot know that they have to terminate their docstrings in certain cases. All they see is that the output type is missing, without knowing why. It is also inconsistent with other places, where terminating the docstring does not require a trailing `\n`.

It also means that the primary regression in 1.7.0 isn't really resolved, because this used to work in 1.6.1 without a trailing `\n`, i.e., the example given in #16486 still comes out wrong, but was working before. Just empirically: In our code base this leads to losing type information in ~100 places or so, because we naturally don't terminate our docstring with `\n` (there is no natural reason to do so), and it will be almost impossible to remind all developers of always remembering this work-around.

Reply:
This has been addressed.   I forgot that pybind11 properties actually have docstrings stored in two locations.